### PR TITLE
Update pubspecs to null-safe dependencies

### PR DESCRIPTION
**Pull request by yjbanov**
Fixes https://github.com/flutter/flutter/issues/69824

### CODE REVIEW

File: lib/web_ui/dev/web_engine_analysis.sh [hunk 1]
```diff
@@ -17,4 +17,6 @@ echo "Running \`pub get\` in 'engine/src/flutter/lib/web_ui'"
 (cd "$WEB_UI_DIR"; $PUB_PATH get)
 
 echo "Running \`dart analyze\` in 'engine/src/flutter/lib/web_ui'"
-(cd "$WEB_UI_DIR"; $DART_PATH analyze --fatal-infos)
+(cd "$WEB_UI_DIR"; $DART_PATH analyze --fatal-infos lib/)
+(cd "$WEB_UI_DIR"; $DART_PATH analyze --fatal-infos test/)
+(cd "$WEB_UI_DIR"; $DART_PATH analyze --fatal-infos dev/)
```
Comment (yjbanov) [inline lib/web_ui/dev/web_engine_analysis.sh:22]: This change is so we don't analyze `build/` and `.dart_tool` directories (they don't exist on CI but they exist locally)

File: lib/web_ui/pubspec.yaml [hunk 2]
```diff
@@ -5,17 +5,17 @@ environment:
...
   analyzer: 0.39.15
   archive: 2.0.13
   http: 0.12.1
   image: 2.1.13
-  js: 0.6.1+1
   mockito: 4.1.1
   path: 1.8.0-nullsafety.3
-  test: 1.14.3
+  test: 1.16.0-nullsafety.9
   quiver: 2.1.3
```
Comment (jakemac53) [inline lib/web_ui/pubspec.yaml:19]: fwiw quiver does now have a null safety release, if you want to use it

**Review by jakemac53 — APPROVED** [review]

File: lib/web_ui/dev/web_engine_analysis.sh (hunk 1 above)
Comment (ditman) [reply lib/web_ui/dev/web_engine_analysis.sh:22]: _(Drive-by review, feel free to ignore)_

Why not `exclude` those two directories in the [analysis_options.yaml](https://github.com/flutter/engine/blob/master/lib/web_ui/analysis_options.yaml) file?

[Syntax reference](https://dart.dev/guides/language/analysis-options#excluding-files).

**Review by nturgut — APPROVED** [review]
LGTM.

As long as the integration tests are also passing, one more measure we can take is patching PR: https://github.com/flutter/engine/pull/22260 and running felt on different modes/build backends.

File: lib/web_ui/dev/web_engine_analysis.sh (hunk 1 above)
Comment (yjbanov) [reply lib/web_ui/dev/web_engine_analysis.sh]: Good find! Done.

File: lib/web_ui/pubspec.yaml (hunk 2 above)
Comment (yjbanov) [reply lib/web_ui/pubspec.yaml]: Done.

**Pull request merged**